lib/ansible/modules/system/etc_hosts.py

```diff
@@ -0,0 +1,429 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils._text import to_bytes
+import os
+import re
+import socket
+import tempfile
+
+
+ANSIBLE_METADATA = {
+    'status': [
+        'preview'
+    ],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: etc_hosts
+author: Jiri Tyr (@jtyr)
+version_added: '2.3'
+short_description: Manages records in the /etc/hosts file
+description:
+  - Add, remove and modify records in the /etc/hosts file.
+options:
+  alias:
+    required: false
+    default: null
+    description:
+      - Alias(es) of the IP address. Multiple aliases can be either a normal
+        YAML list or a string containing comma separated list.
+  hostname:
+    required: false
+    description:
+      - Canonical hostname of the IP address. Required if I(state=present).
+  ip:
+    required: true
+    description:
+      - IP address.
+  path:
+    required: false
+    default: /etc/hosts
+    description:
+      - Path to the hosts file. Required if I(state=present).
+  state:
+    required: false
+    choices: [absent, present]
+    default: present
+    description:
+      - State of the record in the hosts file.
+      - If C(present), I(ip) and I(hostname) are required.
+      - If C(absent), I(ip) or I(hostname) is required.
+
+extends_documentation_fragment:
+  - files
+'''
+
+EXAMPLES = '''
+- name: Add a new record without alias
+  etc_hosts:
+    ip: 10.0.0.1
+    hostname: some.domain1.com
+
+- name: Add a new record with hostname for the same IP
+  etc_hosts:
+    ip: 10.0.0.1
+    hostname: some.domain2.com
+
+- name: Add a new record with another IP for the same hostname
+  etc_hosts:
+    ip: 10.0.0.2
+    hostname: some.domain3.com
+
+- name: Add a new record with hostname and alias for the same IP
+  etc_hosts:
+    ip: 10.0.0.3
+    hostname: some.domain3.com
+    alias: some
+
+- name: Add a new record with multiple aliases
+  etc_hosts:
+    ip: 10.0.0.4
+    hostname: some.domain4.com
+    alias:
+      - some
+      - other
+
+- name: Update the list of aliases for the existing IP and hostname
+  etc_hosts:
+    ip: 10.0.0.4
+    hostname: some.domain4.com
+    alias:
+      - some
+      - other
+      - alias
+
+- name: Remove all records with the specified IP
+  etc_hosts:
+    ip: 10.0.0.1
+    state: absent
+
+- name: Remove all records with the specified hostname
+  etc_hosts:
+    hostname: some.domain3.com
+    state: absent
+
+- name: Remove record with specified IP and hostname
+  etc_hosts:
+    ip: 10.0.0.4
+    hostname: some.domain4.com
+    state: absent
+'''
+
+RETURN = '''
+state:
+  description: state of the target, after execution
+  returned: success
+  type: string
+  sample: "present"
+'''
+
+
+class EtcHosts:
```

Comment:
I would recommend updating to inherit from object.  Just because others did it or are doing it, doesn't necessarily make it a thing we wish to have.

We are trying to improve and standardize on implementations, and the preference here is that we standardize on inheriting from object.

A future task can be to improve modules where this was not followed.